### PR TITLE
Ofnet changes to support new netctl cli

### PR DESCRIPTION
**Pull request by abhinandanpb**
The change contains:
1) decoupling bgp server start from ofnet agent instantiation
2) fixed unit cases to support the changes

### CODE REVIEW

File: ofnetAgent.go [hunk 1]
```diff
@@ -571,18 +570,38 @@ func (self *OfnetAgent) DummyRpc(arg *string, ret *bool) error {
 }
 
 //AddBgpNeighbors add bgp neighbor
-func (self *OfnetAgent) AddBgpNeighbors(As string, peer string) error {
+func (self *OfnetAgent) AddBgp(routerIP string, As string, neighborAs string, peer string) error {
```
Comment (shaleman) [inline ofnetAgent.go]: minor nit: comment does not match new function name

File: ofnetBgp.go [hunk 2]
```diff
@@ -256,6 +288,8 @@ func (self *OfnetBgp) DeleteProtoNeighbor() error {
 //AddProtoNeighbor adds bgp neighbor
 func (self *OfnetBgp) AddProtoNeighbor(neighborInfo *OfnetProtoNeighborInfo) error {
 
+	<-self.start
```
Comment (shaleman) [inline ofnetBgp.go]: Just for my education, why is this required?

File: vlrouter.go [hunk 3]
```diff
@@ -236,6 +248,14 @@ func (self *Vlrouter) RemoveLocalEndpoint(endpoint OfnetEndpoint) error {
 		log.Errorf("Error deleting the endpoint: %+v. Err: %v", endpoint, err)
 	}
 
+	// Remove the endpoint from policy tables
+	if endpoint.EndpointType != "internal-bgp" {
```
Comment (shaleman) [inline vlrouter.go]: Shouldn't this check be there during AddEndpoint() too?

**Comment by shaleman** [comment]
LGTM other than minor comments.

File: ofnetBgp.go (hunk 2 above)
Comment (abhinandanpb) [inline ofnetBgp.go]: To avoid timing issue where the server start is not yet completed  (go routine). neighbor add would come in before that. Found during UT.

File: vlrouter.go (hunk 3 above)
Comment (abhinandanpb) [inline vlrouter.go]: loopback interface endpoint type is internal-bgp and its installed as a local endpoint.

**Comment by shaleman** [comment]
SGTM. Please rebase and we can merge this PR
